src/edu/duke/cs/osprey/gpu/cuda/Gpus.java

```diff
@@ -58,6 +59,18 @@ private Gpus() {
 	}
 	
 	public List<Gpu> getGpus() {
+		
+		// AAO 2017: first sort gpus by decreasing order of free memory as a
+		// crude load-balancing strategy. otherwise, gpus are always hit in
+		// numerical device order for multiple osprey instances, leading to 
+		// unnecessary resource contention.
+		Collections.sort(gpus, new Comparator<Gpu>() {
+			@Override
+			public int compare(Gpu o1, Gpu o2) {
+				return o1.getFreeMemory() > o2.getFreeMemory() ? -1 : 1;
+			}
+		});
```

Comment:
Making the only getGpus() method non-deterministic is a little scary. Maybe it's better to add the load balancing to the Gpu pooling system instead?